apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-tabs.tsx

```diff
@@ -39,18 +39,22 @@ export const InAppTabs = ({ workflow, step }: { workflow: WorkflowResponseDto; s
 
   const { dataSchema, uiSchema, values } = step.controls;
   const schema = useMemo(() => buildDynamicZodSchema(dataSchema ?? {}), [dataSchema]);
-  const newFormValues = useMemo(() => merge(buildDefaultValues(uiSchema ?? {}), values), [uiSchema, values]);
+  const newFormValues = useMemo(
+    () => merge(buildDefaultValues(uiSchema ?? {}), buildDefaultValuesOfDataSchema(dataSchema ?? {}), values),
```

Comment:
Do we need to merge the `uiSchema` defaults with `dataSchema` defaults? 
Here, we should only care about the `uiSchema` defaults.